### ISSUE TITLE
Large number of <option>'s slows down other page functions.

### ISSUE THREAD

**Issue by mrdziuban** [harvesthq/chosen#2158]
I'm running Chrome 39.0.2171.71 on Mac OS X 10.9.5 with Chosen v1.2.0.

Go to [this fiddle](http://jsfiddle.net/nyonmgjc/) which is running an instance of Froala WYSIWYG editor and Chosen for a tags input. Click inside the WYSIWYG editor, type some text, delete it, then try to type again. The editor will lag.

Then go to [this fiddle](http://jsfiddle.net/nyonmgjc/1/) which is identical, but the Chosen initialization is commented out. Do the same thing. The editor will not lag.

Finally, go to [this fiddle](http://jsfiddle.net/nyonmgjc/2/) which is identical to the first one, but there are significantly fewer `<option>`s. Do the same thing. The editor will not lag.

Is there a way to fix this or is it just caused by the large number (nearly 11,000) of options?

**Comment by CodeCollector** [comment]
I ran into a similar problem in IE10 with a list of 900 items, but not in Chrome.  Using the IE profiler showed all the time was being spent setting the innerHtml of the list items that were generated in AbstractChosen.prototype.result_add_option.

I changed:
<code>
option_el.innerText = option.search_text;
</code>
to:
<code>
var searchSplit = $.parseHTML(option.search_text);

        if (searchSplit) {
            for (var index = 0; index < searchSplit.length; index++) {
                option_el.appendChild(searchSplit[index]);
            }
        }
</code>

**Comment by pfiller** [comment]
@mrdziuban Unfortunately, Chosen probably won't realistically perform well when there are 11,000 options. A while back, we rewrote the way Chosen was updating the DOM to make it much faster (see details at #1339), but it only worked up to a certain point (I saw slowdowns starting around 5k items in Chrome).

There are probably plenty of things that could be done to dramatically improve Chosen's speed for super large option lists (not inserting them all into the dom, for one). I suspect all of those things fall well outside the scope of this particular project, though.

@CodeCollector it looks like you might be using the older, slower version. Consider an upgrade and let me know if you're still seeing slowness with 900 items.
